swarm/storage/localstore/gc.go

```diff
@@ -131,27 +128,36 @@ func (db *DB) incGCSize(count int64) {
 	default:
 	}
 	if new >= db.capacity {
-		select {
-		case db.collectGarbageTrigger <- struct{}{}:
-		default:
-		}
+		db.triggerGarbageCollection()
```

Comment:
This approach would create a lot of goroutines while collectGarbage function is running, or it will block incGCSize function until collectGarbage is done. Trigger ensures that only one garbageCollection goroutine (the worker one) as active at one time without blocking other function.